python/Passes.cpp

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "mlir/InitAllTranslations.h"
+#include "ttmlir/Bindings/Python/TTMLIRModule.h"
+#include "ttmlir/RegisterAll.h"
+#include "ttmlir/Target/TTNN/TTNNToFlatbuffer.h"
+
+namespace mlir::ttmlir::python {
+
+void populatePassesModule(py::module &m) {
+  // When populating passes, need to first register them
+
+  mlir::tt::registerAllPasses();
+  mlir::registerAllTranslations();
+  // mlir::tt::ttnn::registerTTNNPipelines();
+
+  m.def("ttir_to_ttnn_backend_pipeline", [](MlirModule module) {
+    mlir::Operation *moduleOp = unwrap(mlirModuleGetOperation(module));
+    mlir::PassManager pm(moduleOp->getName());
+
+    mlir::DialectRegistry registry;
+    mlir::tt::registerAllDialects(registry);
+    mlir::MLIRContext *ctx = unwrap(mlirModuleGetContext(module));
+    ctx->appendDialectRegistry(registry);
+
+    const auto pipeline =
+        mlir::PassPipelineInfo::lookup("ttir-to-ttnn-backend-pipeline");
```

Comment:
Can we just call `createTTIRToTTNNBackendPipeline` directly?

```c++
  TTIRToTTNNBackendPipelineOptions options;
  //.. maybe set options ..
  createTTIRToTTNNBackendPipeline(pm, options);
  if (mlir::failed(pm.run(moduleOp))) {
      throw std::runtime_error("Failed to run pass manager");
  }
```

Reply:
I ran it this way in case there was a more involved method to call Pipeline Options. If it's just as simple as editing the struct then that sounds good.

Reply:
As mentioned offline, this task will be completed later while I focus on getting E2E working with explorer. Refer to #446 for when that gets completed.